### ISSUE TITLE
SARIF: Don't set result level when it doesn't override rule's level

### ISSUE THREAD

**Issue by 3flex** [detekt/detekt#7684]
## Expected Behavior
Result `level` is only set in report output if it overrides the rule's `level`. If it doesn't, it's redundant and can be removed from the output to cut down on the size of the report output.

## Current Behavior
`level` is set on every result.

## Context
Improve SARIF output.

**Comment by VitalyVPinchuk** [comment]
May I clarify, do you mean this part?
![image](https://github.com/user-attachments/assets/d4fff08e-c1e4-4bed-aa07-ab0dd2483a60)

**Comment by 3flex** [comment]
Yes. If a result has the same level as the rule's level then setting the level on the results is redundant - it can be removed so it inherits the level from the rule.

These aren't the right terms as far as the SARIF spec goes but hopefully this makes sense.

**Comment by VitalyVPinchuk** [comment]
May I work on this?

**Comment by 3flex** [comment]
Of course, we love external contributions!

**Comment by BraisGabin** [comment]
@3flex could you explain what you mean with:

> Result `level` is only set in report output if it overrides the rule's `level`.

As far as I can see on our tests we don't set the "rule's `level". So I'm not sure what you mean with that.

**Comment by 3flex** [comment]
Sorry I can't recall all the details, only that it's redundant for the output to contain level on every result if it's already set as the rule level.

I'd fix it myself but it's not a priority.

**Comment by BraisGabin** [comment]
> if it's already set as the rule level.

That's the problem. I can't find where it is defined at the rule level. And I can't fidn that either on the SARIF Spec.

**Comment by VitalyVPinchuk** [comment]
Hi all
Please let me know if you come to a consensus

**Comment by VitalyVPinchuk** [comment]
Oh, now rules don't have severity level...
Instead of default `error` there's `null`
Why?

**Comment by BraisGabin** [comment]
#7835 should unblock this `Issue`

**Comment by marschwar** [comment]
I think @3flex is correct. We should be able to remove the severity on the issue as it will always be the same as the that of the rule. Any circumstance I am missing why the severity of the issue would be different to that of the rule?

I know that we support the notion of rule instances where we instantiate the same rule with different configuration and potentially different level but for the sarif report we consider them to be different rules with the same description and url.

**Comment by BraisGabin** [comment]
After https://github.com/detekt/detekt/pull/7835 this should be easy to implement. We should consider what I wrote on that pr. Right now an issue could have two different severities. Our current implementation will never generate that but our data structures allow it. We should decide if we want to keep the severity iknyñthw issue too.
